src/frontend/src/scheduler/local.rs

```diff
@@ -453,4 +454,39 @@ impl LocalQueryExecution {
             }
         }
     }
+
+    #[inline(always)]
+    fn get_vnode_mapping(&self, table_id: &TableId) -> Option<ParallelUnitMapping> {
+        self.front_env
+            .catalog_reader()
+            .read_guard()
+            .get_table_by_id(table_id)
+            .map(|table| {
+                self.front_env
+                    .worker_node_manager()
+                    .get_fragment_mapping(&table.fragment_id)
+            })
+            .ok()
+            .flatten()
+    }
+
+    fn choose_worker(&self, stage: &Arc<QueryStage>) -> SchedulerResult<Vec<WorkerNode>> {
+        if stage.parallelism.unwrap() == 1 {
+            if let NodeBody::Insert(insert_node) = &stage.root.node
+                && let Some(vnode_mapping) = self.get_vnode_mapping(&insert_node.table_id.into()) {
+                    let worker_node = {
+                        let parallel_unit_ids = vnode_mapping.iter_unique().collect_vec();
+                        let candidates = self.front_env
+                            .worker_node_manager()
+                            .get_workers_by_parallel_unit_ids(&parallel_unit_ids)?;
+                        candidates.choose(&mut rand::thread_rng()).unwrap().clone()
+                    };
+                    Ok(vec![worker_node])
+            } else {
+                Ok(vec![self.front_env.worker_node_manager().next_random()?])
+            }
+        } else {
+            Ok(self.front_env.worker_node_manager().list_worker_nodes())
```

Comment:
Hmm should we list all worker nodes here?
What if parallelism = 3, and worker nodes = 5, does that mean we choose 5 workers?
Should it match exactly?

Reply:
I think it should be workers with target table only.

Reply:
We have process case that the stage have scan node before, so in this case the stage don't have the scan node, I think we just need to randomlly select N parallelism worker?🤔

(Previous process is list worker node)
https://github.com/risingwavelabs/risingwave/blob/1ad23babf8b61e89cbe9ffc3a8fadb7f878f9ffe/src/frontend/src/scheduler/local.rs#L325